server/protocols/webadmin/webadmin-cassandra/src/test/java/org/apache/james/webadmin/routes/CassandraMigrationRoutesTest.java

```diff
@@ -175,12 +186,23 @@ public void postShouldDoMigrationToNewVersion() throws Exception {
     public void postShouldNotDoMigrationWhenCurrentVersionIsNewerThan() throws Exception {
         when(schemaVersionDAO.getCurrentSchemaVersion()).thenReturn(CompletableFuture.completedFuture(Optional.of(CURRENT_VERSION)));
 
-        given()
+        Map<String, Object> errors = given()
             .body(String.valueOf(OLDER_VERSION))
         .with()
             .post("/upgrade")
         .then()
-            .statusCode(410);
+            .statusCode(410)
+            .contentType(ContentType.JSON)
+            .extract()
+            .body()
+            .jsonPath()
+            .getMap(".");
+
+        assertThat(errors)
+            .containsEntry("statusCode", 410)
+            .containsEntry("type", "WrongState")
+            .containsEntry("message", "The migration requested can not be performed")
+            .containsEntry("cause", "Current version is already up to date");
```

Comment:
are you sure ? then why it says "gone" ?

Reply:
Good question

Reply:
Hum I think the logic behind that it was that this request was available upon a time.
Because, in rest everything is a resource, this request is represented by a ressource that is not here anymore.
But I think BAD_REQUEST will be more appropriate I am going to change for that

Reply:
the problem is that upgrade is not a resource.